com.unity.netcode.gameobjects/Runtime/Core/NetworkManager.cs

```diff
@@ -86,6 +86,12 @@ public NetworkPrefabHandler PrefabHandler
         private bool m_ShuttingDown;
         private bool m_StopProcessingMessages;
 
+        // <summary>
+        // When disconnected from the server, the server may send a reason. If a reason was sent, this property will
+        // tell client code what the reason was. It should be queried after the OnClientDisconnectCallback is called
+        // </summary>
+        public string DisconnectReason { get; internal set; }
```

Comment:
Not a huge fan of a member variable on NM (not least of which because you're not cleaning it up on subsequent starts).

I think this is part of a bigger issue I have with this approach. I'd much prefer to see the reason combined with the actual disconnect instead of a separate message that has no effect/impact prior to disconnect...

Reply:
```
        private void Initialize(bool server)
        {
[...]
            DisconnectReason = "";
```

Which is called on StartHost, StartClient and StartServer